libcxx/include/__ranges/iota_view.h

```diff
@@ -374,10 +375,11 @@ namespace views {
 namespace __iota {
 struct __fn {
   template <class _Start>
+    requires(requires(_Start __s) { ranges::iota_view<decay_t<_Start>>(std::forward<_Start>(__s)); })
```

Comment:
I think we should do choose one of the following approaches:
- exactly using `decltype(ranges::iota_view<decay_t<_Start>>(std::forward<_Start>(__start)))` to achieve expression-equivalency, or
- using `requires` and deduced return type (plain `auto` for this `operator()`).

In either case, we should use `noexcept(noexcept(ranges::iota_view<decay_t<_Start>>(std::forward<_Start>(__start))))` for consistency.

(However, it is currently always equivalent to `noexcept(false)`. No change requested for exception specification strengthening.)

Reply:
> I believe I have implemented the second approach already or do I misunderstand you, could you please clarify?

But the (non-deduced) trailing return type isn't removed.

Reply:
Ups! Sorry! Done! Thank you!